core/src/main/java/org/apache/iceberg/actions/SizeBasedFileRewriter.java

```diff
@@ -250,6 +252,11 @@ protected long writeMaxFileSize() {
     return (long) (targetFileSize + ((maxFileSize - targetFileSize) * 0.5));
   }
 
+  /** Output spec id rewriter to use. Default to current spec id */
```

Comment:
nit: why this is function here and not with the outputSpecId(config) method?

Reply:
nvm, I think it probably makes more sense here. Although the javadoc comment is not accurate. This code only returns the outputSpecId saved by this class. The Default logic is in the other function.